pkg/sql/lease.go

```diff
@@ -574,20 +599,65 @@ func (t *tableState) acquire(
 ) (*tableVersionState, error) {
 	t.mu.Lock()
 	defer t.mu.Unlock()
-	// Wait for any existing lease acquisition.
-	t.acquireWait()
 
-	// Acquire a lease if no lease exists or if the latest lease is
-	// about to expire.
+	// When a lease is about to expire but has not yet (determined by
+	// leaseHasEnoughLifeLeft), then ensure an asynchronous routine to renew the
+	// lease is started.
+	if s := t.mu.active.findNewest(); s != nil && !s.hasExpired(timestamp) && !t.mu.asyncRenewalInProgress && !s.leaseHasEnoughLifeLeft() {
+		// Set a flag denoting async acquisition has begun to prevent this from
+		// happening multiple times.
+		t.mu.asyncRenewalInProgress = true
+		go t.acquireIfRenewalNotInProgress(m)
```

Comment:
use t.stopper.RunAsyncTask() to prevent goroutines leaks in tests,  but as I said above you want to call this from within acquireNodeLease()